### PR TITLE
[Async Message]Preview views

### PR DESCRIPTION
**Pull request by ipavlidakis**
**Stack**:
- #2623
- #2622
- #2621
- #2620 ⬅
- #2619
- #2618

⚠️ *Part of a stack created by [spr](https://github.com/ejoffe/spr). Do not merge manually using the UI - doing so may have unexpected results.*

### CODE REVIEW

File: Sources/StreamChatUI/CommonViews/Attachments/AttachmentViews/VoiceRecordingAttachmentComposerPreview.swift [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+
+    open private(set) lazy var fileNameAndDurationStack: ContainerStackView = {
+        let stack = ContainerStackView(arrangedSubviews: [fileNameLabel, durationLabel])
+        stack.axis = .vertical
+        stack.alignment = .leading
+        stack.spacing = 3
+        return stack
+            .withoutAutoresizingMaskConstraints
+            .withAccessibilityIdentifier(identifier: "fileNameAndDurationStack")
+    }()
```
Comment (nuno-vieira) [inline Sources/StreamChatUI/CommonViews/Attachments/AttachmentViews/VoiceRecordingAttachmentComposerPreview.swift:82]: As mentioned in a previous PR, usually it is best if we provide a more generic name for the stack 👍

**Review by nuno-vieira — APPROVED** [review]
Just added a small comment, other than that LGTM! ✅

File: Sources/StreamChatUI/CommonViews/Attachments/AttachmentPreviewProvider.swift [hunk 2]
```diff
@@ -49,3 +49,19 @@ extension VideoAttachmentPayload: AttachmentPreviewProvider {
         return preview
     }
 }
+
+extension VoiceRecordingAttachmentPayload: AttachmentPreviewProvider {
+    public static var preferredAxis: NSLayoutConstraint.Axis { .vertical }
```
Comment (polqf) [inline Sources/StreamChatUI/CommonViews/Attachments/AttachmentPreviewProvider.swift:54]: What's the point of placing this here? Clients will not be able to override it

File: Sources/StreamChatUI/CommonViews/Attachments/AttachmentViews/VoiceRecordingAttachmentComposerPreview.swift [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+        .withoutAutoresizingMaskConstraints
+
+    /// A button that can be used to control the playback of the VoiceRecording.
+    open private(set) lazy var playPauseButton: PlayPauseButton = .init()
+        .withoutAutoresizingMaskConstraints
+        .withAccessibilityIdentifier(identifier: "actionButton")
+
+    /// The container that the fileNameLabel and the durationLabel will be added into.
+    open private(set) lazy var centerContainerStackView: ContainerStackView = {
+        let stack = ContainerStackView(arrangedSubviews: [fileNameLabel, durationLabel])
```
Comment (polqf) [inline Sources/StreamChatUI/CommonViews/Attachments/AttachmentViews/VoiceRecordingAttachmentComposerPreview.swift:66]: It looks a bit weird to me that we have layout logic here and in setUpLayout

File: Sources/StreamChatUI/CommonViews/Attachments/AttachmentPreviewProvider.swift (hunk 2 above)
Comment (nuno-vieira) [reply Sources/StreamChatUI/CommonViews/Attachments/AttachmentPreviewProvider.swift:54]: This is part of the `AttachmentPreviewProvider` protocol

**Comment by ipavlidakis** [comment]
✓ Commit merged in pull request [#2623](https://github.com/GetStream/stream-chat-swift/pull/2623)

**Pull request closed without merging**